database/query/eval_history.sql

```diff
@@ -93,3 +93,71 @@ INSERT INTO alert_events(
     $3,
     $4
 );
+
+-- name: ListEvaluationHistory :many
+SELECT s.id::uuid AS evaluation_id,
+       s.most_recent_evaluation as evaluated_at,
+       -- entity type
+       CASE WHEN ere.repository_id IS NOT NULL THEN 'repository'::entities
+            WHEN ere.pull_request_id IS NOT NULL THEN 'pull_request'::entities
+            WHEN ere.artifact_id IS NOT NULL THEN 'artifact'::entities
+       END AS entity_type,
+       -- entity id
+       CASE WHEN ere.repository_id IS NOT NULL THEN r.id
+            WHEN ere.pull_request_id IS NOT NULL THEN pr.id
+            WHEN ere.artifact_id IS NOT NULL THEN a.id
+       END AS entity_id,
+       -- entity name
+       CASE WHEN ere.repository_id IS NOT NULL THEN r.repo_name
```

Comment:
Shouldn't `entity_name` for repositories be `CONCAT(r.repo_owner, '/', r.repo_name)`?

Reply:
I'll probably move that in the upper layer, I'm not particularly fond of having this sort of things in queries, as one can introduce very subtle bugs by changing just one line.

That said, I think this issues should be solved on the write path, by writing something like an extra `calculated_name` column, so that provider-specific logic can run before any data reaches storage.